builds/fabric8-che/assembly/user-auth-valve/src/main/java/com/redhat/che/valve/UserAuthValve.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat inc.
+
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+
+package com.redhat.che.valve;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.catalina.connector.Request;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.keycloak.KeycloakSecurityContext;
+import org.keycloak.adapters.tomcat.KeycloakAuthenticatorValve;
+
+/**
+ * Performs Keycloak and OpenShift.io user validation. Prompts user to login if necessary
+ * and checks if the logged in user has access to the current Che instance.
+ *
+ * @see KeycloakAuthenticatorValve
+ *
+ * @author amisevsk
+ */
+public class UserAuthValve extends KeycloakAuthenticatorValve {
```

Comment:
Reading the class name and the description, it's not clear what makes `UserAuthValve` different from `KeycloakAuthenticatorValve`. maybe `CheOsioUserValidationAuthValve`